### PR TITLE
Making vaultwarden run in linux/386

### PR DESCRIPTION
**Pull request by amedeorizzo**
I was trying to make vaultwarden work on a 32-bit linux server of a friend. After some tries with the official image I decided to build a docker image specifically for linux/386 systems. These are the changes I made.
Builded image can be found here: https://hub.docker.com/r/amedeorizzo/vaultwarden386

### CODE REVIEW

File: docker/Dockerfile.debian [hunk 1]
```diff
@@ -193,4 +194,4 @@ COPY --from=build /app/target/final/vaultwarden .
 
 HEALTHCHECK --interval=60s --timeout=10s CMD ["/healthcheck.sh"]
 
-CMD ["/start.sh"]
+CMD ["sh", "/start.sh"]
```
Comment (williamdes) [inline docker/Dockerfile.debian:197]: is this change really required for 386 ?

**Comment by amedeorizzo** [comment]
> This will totally break our other builds, so it will not be implemented like this in the current way.
> 
> I also have not seen any requests before to build for this architecture, so not sure if we should try to add support for it in our repo. I'm not fully against it, and i might be able to add this in a better way so you do not have to make these changes all the time.
> 
> But, it will be landing on the not fully supported list.

Yes, I'm aware that needs to be implemented in another way to not break the other builds. This could be an idea for the future.

File: docker/Dockerfile.debian (hunk 1 above)
Comment (williamdes) [reply docker/Dockerfile.debian:197]: Let's trim down this patch to get it merged

Comment (BlackDex) [reply docker/Dockerfile.debian:197]: It won't be merged like this hehe.
Should be handled totally different.

But I'm fine with this for now to keep it as a reminder.

**Comment by Ornias1993** [comment]
> This will totally break our other builds, so it will not be implemented like this in the current way.
> 
> I also have not seen any requests before to build for this architecture, so not sure if we should try to add support for it in our repo. I'm not fully against it, and i might be able to add this in a better way so you do not have to make these changes all the time.
> 
> But, it will be landing on the fully supported list.

I personally see no usecase for this, is there even any x86 non x86_64 hardware being released anymore?

**Comment by BlackDex** [comment]
> > This will totally break our other builds, so it will not be implemented like this in the current way.
> > I also have not seen any requests before to build for this architecture, so not sure if we should try to add support for it in our repo. I'm not fully against it, and i might be able to add this in a better way so you do not have to make these changes all the time.
> > But, it will be landing on the fully supported list.
> 
> I personally see no usecase for this, is there even any x86 non x86_64 hardware being released anymore?

That doesn't matter of course, I personally also so no use-case for SSO for Vaultwarden, that doesn't mean it might be useful for others.

I'm not against adding a simple way to allow building it. We will just not give support for it. If it doesn't work, it just doesn't work. Same currently for MIPS, might be someone is using that already in some way, but we will not support it as one of the main architectures i think.

**Comment by BlackDex** [comment]
i386/i686 is still supported by Debian until at least 2028

**Comment by Ornias1993** [comment]
> i386/i686 is still supported by Debian until at least 2028

The important note on this, is that Debian has current customers/clients/users using it, because they offered it before.
I doubt they would add support for it in 2024 either, if it wasn't already there...

**Comment by BlackDex** [comment]
Closing this already in favor of #4524
